app/serializers/api/participant_serializer.rb

```diff
@@ -0,0 +1,63 @@
+module API
+  class ParticipantSerializer < Blueprinter::Base
+    identifier :ecf_user_id, name: :id
+    field(:type) { "participant" }
+
+    class AttributesSerializer < Blueprinter::Base
+      exclude :id
+
+      view :v3 do
+        field(:full_name) { |t, _| Teachers::Name.new(t).full_name }
+        field(:trn) { |t, _| t.trn }
+
+        field(:ecf_enrolments) do |teacher, _options|
+          (teacher.ect_at_school_periods + teacher.mentor_at_school_periods).map do |school_period|
+            # Better using uuids instead as we don't know whether latest_training_period is an id from ect_at_school_periods or mentor_at_school_periods
+            training_period = school_period.training_periods.find { |training_period| teacher.latest_training_period.include?(training_period.id) }
+
+            next if training_period.blank?
```

Comment:
fyi for anyone looking at this in the future
this will create nil value in the `map` result.